code/__DEFINES/dcs/signals.dm

```diff
@@ -345,8 +345,10 @@
 
 #define COMSIG_XENOMORPH_HEADBITE "headbite"
 
+#define COMSIG_XENOMORPH_DEATH "xenomorph_death"
 #define COMSIG_XENOMORPH_GIBBING "xenomorph_gibbing"
 #define COMSIG_XENOMORPH_POSTEVOLVING "xenomorph_evolving"
+#define COMSIG_XENOMORPH_ABILITY_ON_PRE_UPGRADE "xenomorph_pre_upgrade"
```

Comment:
why are you adding these and where are they used

Reply:
Death is used to remove the warp shadow on the death of the Wraith.

Pre-upgrade was used to preserve vars between upgrades, but is probably no longer necessary so I can remove this